sped_purchase/models/sped_documento.py

```diff
@@ -80,26 +80,29 @@ def purchase_order_change(self):
     def _onchange_allowed_purchase_ids(self):
         result = {}
 
-        # A PO can be selected only if at least one PO line is not already in the invoice
-        purchase_line_ids = self.item_ids.mapped('purchase_line_ids')
-        purchase_ids = self.item_ids.mapped('purchase_ids').filtered(lambda r: r.order_line <= purchase_line_ids)
-
-        result['domain'] = {'purchase_order_ids': [
-            ('invoice_status', '=', 'to invoice'),
-            ('id', 'not in', purchase_ids.ids),
-        ]}
-
-        if self.participante_id:
-            result['domain']['purchase_order_ids'].append(
-                ('participante_id', '=', self.participante_id.id)
-            )
-
-        if self.empresa_id:
-            result['domain']['purchase_order_ids'].append(
-                ('empresa_id', '=', self.empresa_id.id)
-            )
+    @api.multi
+    def write(self, vals):
+        res = super(SpedDocumento, self).write(vals)
+        for documento in self:
```

Comment:
Se for pra alterar o comportamento depois de chamar o super, acho que deveria iterar o res e não o self.